DDPB-2512 - NDR report incorrectly displayed

acsauk-mojd opened the pull request:
I've moved the NDR creation step to the Lay Deputy specific steps during the upsert action to ensure they are only created for Lay Deputies.

As I was running the unit tests locally I encountered an error while loading the fixtures class so I've namespaced this and updated references as part of this PR.

aiesoftware commented on src/AppBundle/Controller/ClientController.php:
```diff
@@ -49,6 +49,13 @@ public function upsertAction(Request $request)
 
         if ($this->getUser()->isLayDeputy()) {
+            //add NDR if not added yet
+            // TODO move to listener or service
```
I think we should remove these comments..

acsauk-mojd commented:
The requirements of this ticket evolved slightly - NDRs should only be created for Lays if the NDR Enabled checkbox has been ticked from the admin side. We were previously just checking if an NDR existed for the client rather than checking for the box being checked. I've updated the logic to reflect this and added a couple fo extra assertions for this in the unit tests.

slizzio commented on src/AppBundle/Controller/ClientController.php:
```diff
@@ -60,14 +71,8 @@ public function upsertAction(Request $request)
-            $ndr = new EntityDir\Ndr\Ndr($client);
-            $this->persistAndFlush($ndr);
-        }
+        $em->persist($client);
```
check client is set before persisting

slizzio commented on tests/AppBundle/Controller/ClientControllerTest.php:
```diff
@@ -121,10 +121,9 @@ public function testupsertPostLayDeputy()
     }
 
-    public function testupsertPut()
+    public function testupsertPut_lay_deputy()
```
camelCase?

acsauk-mojd replied:
I used to be exclusively camel case for test function names until my last role where we kept the actual function it was testing in camel case but gave the scenario/detail using snake case. Happy to change if thats the preferred style here though!

slizzio commented on tests/AppBundle/Controller/ClientControllerTest.php:
```diff
@@ -148,6 +147,30 @@ public function testupsertPut()
         $this->assertEquals(self::$deputy1->getId(), $client->getUsers()->first()->getId());
+    }
+
+    public function testupsertPut_lay_deputy_ndr_enabled()
```
camelCase

slizzio commented on tests/AppBundle/Controller/ClientControllerTest.php:
```diff
@@ -148,6 +147,30 @@ public function testupsertPut()
+        $this->assertInstanceOf(Ndr::class, $client->getNdr());
+    }
+
+    public function testupsertPut_PA()
```
camelcase
